decidim-budgets/spec/types/integration_schema_spec.rb

```diff
@@ -188,4 +190,233 @@
       expect(response["participatoryProcess"]["components"].first["budget"]).to eq(budget_single_result)
     end
   end
+
+  context "with resource visibility" do
```

Comment:
Budgets have a different behavior. The projects are listed only if the budget is visible, but that visibility is checking to see if component and space are published. I had to go on the wider route.